### PR TITLE
CI: Requires Lint to pass before running other jobs. Allow CI to be skipped for non code changes

### PR DESCRIPTION
**Pull request by sithlord48**
- Adjust CI so that all jobs need linting to pass to run . 
 - CI will not run if the PR only contains changes to these files:
   - any md files  (README etc..)
   -`.gitignore`
   -`.gitattributes` 
   - github `ISSUES_TEMPLATES`
   -  `.env.example`
   -  the `cspell.json` file

### CODE REVIEW

**Review by nbolton — APPROVED** [review]
Alright, let's give this a try.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,6 +15,13 @@ on:
       - reopened
       - synchronize
       - ready_for_review
+    paths-ignore:
+      - '**/*.md'
+      - '.env-example'
+      - '.gitignore'
+      - 'Brewfile'
+      - 'Chocolatey.config'
```
Comment (nbolton) [inline .github/workflows/ci.yml:23]: Not sure about this. CI uses these so I think triggering CI would be wise...

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -5,6 +5,27 @@ on:
...
+      - '!.github/workflow/codeql-analysys.yml'
+      - 'doc/**'
+      - 'res/**'
+      - '!res/config.h.in'
+      - 'scripts/**'
+      - '!scripts/install_deps.py'
+      - '.env-example'
+      - '.gitignore'
+      - 'Brewfile'
+      - 'Chocolatey.config'
```
Comment (nbolton) [inline .github/workflows/codeql-analysis.yml:21]: Same comment as above.

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -5,6 +5,27 @@ on:
   workflow_call:
   push:
     branches: [master]
+    paths-ignore:
+      - '.clang-format'
```
Comment (nbolton) [inline .github/workflows/codeql-analysis.yml:9]: Ah, I'm certain this should trigger CI change.

**Review by nbolton — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sithlord48) [reply .github/workflows/ci.yml:23]: I was wondering that these (and lot of other stuff are doing

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (sithlord48) [reply .github/workflows/codeql-analysis.yml:9]: yes but, it should not trigger a codeql change when a pr is landed into master

**Comment by nbolton** [comment]
> Minor CI Changes 

Some of these proposed changes are actually quite major. I think that we should avoid using subjective PR titles; we should describe the change rather than summarizing in a subjective/ambiguous way.

File: .github/workflows/sonarcloud-analysis.yml [hunk 4]
```diff
@@ -8,6 +8,25 @@ on:
         required: true
   push:
     branches: [master]
+    paths-ignore:
+      - '.clang-format'
```
Comment (nbolton) [inline .github/workflows/sonarcloud-analysis.yml:12]: Same as above, quite a major change.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (sithlord48) [reply .github/workflows/codeql-analysis.yml:21]: unless codeQL is building on windows on mac these will not affect this jobs run

File: .github/workflows/sonarcloud-analysis.yml [hunk 5]
```diff
@@ -8,6 +8,25 @@ on:
...
+      - '**/*.md'
+      - '.github/**'
+      - '!.github/workflow/sonarcloud-analysys.yml'
+      - 'doc/**'
+      - 'res/**'
+      - '!res/config.h.in'
+      - 'scripts/**'
+      - '!scripts/install_deps.py'
```
Comment (nbolton) [inline .github/workflows/sonarcloud-analysis.yml:20]: All script changes should be tested in CI. I really don't think it's a good idea to ignore these changes.

File: .github/workflows/sonarcloud-analysis.yml [hunk 6]
```diff
@@ -8,6 +8,25 @@ on:
         required: true
   push:
     branches: [master]
+    paths-ignore:
+      - '.clang-format'
+      - '**/*.md'
+      - '.github/**'
```
Comment (nbolton) [inline .github/workflows/sonarcloud-analysis.yml:14]: Not sure I quite understand why this would be a good thing.

File: .github/workflows/sonarcloud-analysis.yml (hunk 5 above)
Comment (sithlord48) [reply .github/workflows/sonarcloud-analysis.yml:20]: This is only code analsys when landed into master

File: .github/workflows/sonarcloud-analysis.yml (hunk 6 above)
Comment (sithlord48) [reply .github/workflows/sonarcloud-analysis.yml:14]: we dont' need to run this job if only those items changed they are not scanned at all

File: .github/workflows/sonarcloud-analysis.yml (hunk 4 above)
Comment (sithlord48) [reply .github/workflows/sonarcloud-analysis.yml:12]: no its not this file is not even scanned. code format changes are not going to change those results so no need to run the job and waste ci time for this file change

**Comment by nbolton** [comment]
Converted to draft to reduce notification noise. Let me know if you need it to be open for some reason.

File: .github/workflows/sonarcloud-analysis.yml (hunk 4 above)
Comment (nbolton) [reply .github/workflows/sonarcloud-analysis.yml:12]: Ah, yeah just noticed the workflow name, never mind.

**Comment by nbolton** [comment]
> TL;DR: I would really appreciate a more descriptive PR title.

@sithlord48 In case you missed it ^

**Review by nbolton — APPROVED** [review]

**Review by nbolton — APPROVED** [review]
